Makefile

```diff
@@ -317,17 +317,12 @@ test-python-universal:
 	FEAST_USAGE=False IS_TEST=True python -m pytest -n 8 --integration sdk/python/tests
 
 format-python:
-	# Sort
-	cd ${ROOT_DIR}/sdk/python; python -m isort feast/ tests/
-
-	# Format
-	cd ${ROOT_DIR}/sdk/python; python -m black --target-version py38 feast tests
+	cd ${ROOT_DIR}/sdk/python; python -m ruff check --fix feast/ tests/
+	cd ${ROOT_DIR}/sdk/python; python -m ruff format feast/ tests/
 
 lint-python:
-	cd ${ROOT_DIR}/sdk/python; python -m mypy feast
-	cd ${ROOT_DIR}/sdk/python; python -m isort feast/ tests/ --check-only
-	cd ${ROOT_DIR}/sdk/python; python -m flake8 feast/ tests/
-	cd ${ROOT_DIR}/sdk/python; python -m black --check feast tests
+	# cd ${ROOT_DIR}/sdk/python; python -m mypy feast
+	cd ${ROOT_DIR}/sdk/python; ruff check feast/ tests/
```

Comment:
I meant this line
ruff has 2 steps to fully conduct lint check, `ruff check` and `ruff format --check`
I know it's a bit confusing but they are planning to unify these two commands

Reply:
I don't think they behave the same, e.g. the error codes defined in the `select` key of the `tool.ruff.lint` configuration table are not taken into account by `ruff format --check`.

For instance, if I create `feast/my.py` as:
```python
import itertools
import copy
```
Then `ruff check` detects the error:
```console
$ python -m ruff check feast/my.py 
warning: One or more modules are part of multiple import sections, including: `feast`
feast/my.py:1:1: I001 [*] Import block is un-sorted or un-formatted
feast/my.py:1:8: F401 [*] `itertools` imported but unused
feast/my.py:2:8: F401 [*] `copy` imported but unused
```
but `ruff format --check` doesn't:
```console
$ ruff format --check feast/my.py 
warning: One or more modules are part of multiple import sections, including: `feast`
1 file already formatted
```

Reply:
yes they do different jobs, `ruff check` will take care of isort stuff while `ruff format --check` can do code format similar to `black`
since I see we're replacing black in this PR so we need to handle `ruff format` as well

Reply:
> since I see we're replacing black in this PR so we need to handle `ruff format` as well

can you look at the latest changes? it should be there in the `format-python` target, IIUC you point

Reply:
yeah sorry didn't declare clearly, I meant [this](https://arc.net/l/quote/ywotfszk)

running `ruff format --check` will help checking format without touching the code, I believe we should add this to `lint-python` as well

Reply:
let me add it then, I misunderstood